qiskit_nature/algorithms/pes_samplers/bopes_sampler.py

```diff
@@ -90,18 +97,26 @@ def __init__(
                     "num_bootstrap must be None or an integer greater than or equal to 2"
                 )
 
-        if isinstance(self._gss.solver, VariationalAlgorithm):  # type: ignore
+        if isinstance(self._solver_wrapper.solver, VariationalAlgorithm):
             # Save initial point passed to min_eigensolver;
             # this will be used when NOT bootstrapping
-            self._initial_point = self._gss.solver.initial_point  # type: ignore
+            self._initial_point = self._solver_wrapper.solver.initial_point
 
-    def sample(self, problem: BaseProblem, points: List[float]) -> BOPESSamplerResult:
+    def sample(
+        self,
+        problem: BaseProblem,
+        points: List[float],
+        aux_operators: Optional[
+            ListOrDictType[Union[SecondQuantizedOp, PauliSumOp, Callable]]
```

Comment:
We should document the expectations for this `Callable`. You should do that in two ways:
1. add a more detailed type hint
2. add an explanation to the docstring below